docs/API/metadata.md

```diff
@@ -7,9 +7,30 @@ authenticated user, while other metadata is maintained internally by the server
 and can't be changed. Authenticated users can also add any additional metadata
 that might be of use.
 
->__NOTE__: right now the ability to search and filter using metadata is
-limited, but our intent is to be able to use any defined metadata value both
-for searches and to filter the results of [datasets/list](V1/list.md).
+Dataset metadata is represented as a set of nested JSON objects. There are four
+distinct key namespaces. These can be addressed (read or changed) at any level
+of the hierarchy using a dotted name path, for example `dataset.resource_id`
+for a dataset's resource ID, or `global.environment.cluster.ip`. The keys are
+lowercase alphabetic, plus digits, hyphen, and underscore: so `global.u-7` is
+OK, but `global.Frank` isn't.
```

Comment:
I think (or at least intended to think) that I mention this in filtering, and I'd rather keep it local rather than pollute the generic information here.

We need to document the tarball format and `metadata.log`, and that page (which we could link here) would be a better venue.